plugin/storage/factory_config.go

```diff
@@ -47,39 +52,48 @@ type FactoryConfig struct {
 //
 // For backwards compatibility it also parses the args looking for deprecated --span-storage.type flag.
 // If found, it writes a deprecation warning to the log.
-func FactoryConfigFromEnvAndCLI(args []string, log io.Writer) FactoryConfig {
+func FactoryConfigFromEnvAndCLI(args []string, logger *zap.Logger) FactoryConfig {
 	spanStorageType := os.Getenv(SpanStorageTypeEnvVar)
 	if spanStorageType == "" {
 		// for backwards compatibility check command line for --span-storage.type flag
-		spanStorageType = spanStorageTypeFromArgs(args, log)
+		spanStorageType = spanStorageTypeFromArgs(args, logger)
 	}
 	if spanStorageType == "" {
 		spanStorageType = cassandraStorageType
 	}
-	depStoreType := os.Getenv(DependencyStorageTypeEnvVar)
-	if depStoreType == "" {
-		depStoreType = spanStorageType
+	spanWriterTypes := strings.Split(spanStorageType, ",")
+
+	spanReaderType := os.Getenv(SpanReaderTypeEnvVar)
+	if spanReaderType == "" {
+		if len(spanWriterTypes) > 1 {
+			logger.Warn("the first span storage type listed (" + spanWriterTypes[0] + ") will be used for reading. " +
+				"Please use environment variable " + SpanReaderTypeEnvVar + " to specify which storage type to read from")
+		}
+		spanReaderType = spanWriterTypes[0]
+	}
+
+	depStorageType := os.Getenv(DependencyStorageTypeEnvVar)
+	if depStorageType == "" {
+		depStorageType = spanReaderType
 	}
 	return FactoryConfig{
-		SpanStorageType:         spanStorageType,
-		DependenciesStorageType: depStoreType,
+		SpanWriterTypes:         spanWriterTypes,
+		SpanReaderType:          spanReaderType,
+		DependenciesStorageType: depStorageType,
+		logger:                  logger,
 	}
 }
 
-func spanStorageTypeFromArgs(args []string, log io.Writer) string {
+func spanStorageTypeFromArgs(args []string, logger *zap.Logger) string {
 	for i, token := range args {
 		if i == 0 {
 			continue // skip app name; easier than dealing with +-1 offset
 		}
 		if !strings.HasPrefix(token, spanStorageFlag) {
 			continue
 		}
-		fmt.Fprintf(
-			log,
-			"WARNING: found deprecated command line option %s, please use environment variable %s instead\n",
-			token,
-			SpanStorageTypeEnvVar,
-		)
+		logger.Warn("found deprecated command line option " + token +
```

Comment:
sorry, I didn't realize we already use this io.Writer. Let's continue using it instead of creating redundant logger in all mains.